cpp/lib/query/query_manager.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "typedb/query/query_manager.hpp"
+#include "typedb/common/error_message.hpp"
+#include "typedb/common/exception.hpp"
+#include "typedb/common/future.hpp"
+#include "typedb/connection/transaction.hpp"
+
+#include "typedb/concept/value/value.hpp"
+
+#include "../common/macros.hpp"
+#include "../common/native.hpp"
+
+#define CHECK_QUERY(QUERY)                                                    \
+    {                                                                         \
+        if ("" == (QUERY)) throw Utils::exception(QueryError::MISSING_QUERY); \
+    }
+
+namespace TypeDB {
+
+QueryManager::QueryManager(TypeDB::Transaction* transaction)
+    : transaction(transaction) {}
+
+VoidFuture QueryManager::define(const std::string& query, const Options& options) const {
+    CHECK_NATIVE(transaction);
+    CHECK_QUERY(query);
+    WRAPPED_NATIVE_CALL(VoidFuture, _native::query_define(transaction->getNative(), query.c_str(), options.getNative()));
+}
```

Comment:
The UX here to pass options every time isn't great. Should we introduce a variant that takes the transaction options by default?